spring-cloud-commons/src/main/java/org/springframework/cloud/client/ServiceInstance.java

```diff
@@ -71,4 +73,18 @@ default String getScheme() {
 		return null;
 	}
 
+	/**
+	 * @return The raw weight of the service instance.
+	 */
+	default int getWeight() {
```

Comment:
New interface methods should not be necessary - we should rely on metadata entries.

Reply:
But there is a problem if using metadata directly, weight is a number but the value type of metadata is String, so we will do unnecessary `Integer.parseInt` cause each time we will get the same value.